### PR TITLE
Project whitelist for users

### PR DESCRIPTION
**Pull request by YLiLarry**
Projects whitelist is in ~cs136/assignment_skeletons/project_whitelist.json
User whitelist is ~cs136/assignment_skeletons/user_whitelist.json

I also added their cgis in ~cs136/cgi-bin/. Not sure how to execute them tho.

I still have cross domain request issue when I try to test it on ugister, so alternatively I tested by imitating the server response. It should work on the live server.

### CODE REVIEW

**Comment by e45lee** [comment]
Try wrapping the JSON files in a cgi-script that outputs "Access-Control-Allow-Origin: *".  You will have to edit .htaccess.

**Comment by YLiLarry** [comment]
Hi Edward, don't I have to restart the https://www.student.cs.uwaterloo.ca/ apache server to make that effective?

File: src/frontend/js/project-service.js [hunk 1]
```diff
@@ -34,6 +34,8 @@ angular.module('seashell-projects', ['seashell-websocket', 'marmoset-bindings'])
       // TODO: update with real template path.
       var PROJ_ZIP_URL_TEMPLATE = SKEL_ROOT_URL + "%s-seashell.zip";
       var PROJ_FILE_LIST_URL_TEMPLATE = CGI_URL + "skeleton_file_list.rkt?template=%s";
+      var PROJ_WHITE_LIST_URL = SKEL_ROOT_URL + "project_whitelist.json";
```
Comment (e45lee) [inline src/frontend/js/project-service.js]: Shouldn't this be a CGI script?

**Comment by e45lee** [comment]
No; also, you should run the HTTP request against the CGI script.

**Comment by YLiLarry** [comment]
Yeah I will change them to cgi scripts.
Can you take a look at why https://www.student.cs.uwaterloo.ca/~cs136/cgi-bin/user_whitelist.cgi is displayed as a plain text?
I'm not sure how the server is configured. yet.

**Comment by YLiLarry** [comment]
:3 
Looked into `cat /etc/apache2/sites-available/default` but nothing useful found.

**Comment by Gibstick** [comment]
I fixed the .htaccess in ~cs136/public_html/cgi-bin/ and it executes now.

**Comment by YLiLarry** [comment]
How did you do that? What did you change... :3

**Comment by Gibstick** [comment]
You have to specifically whitelist every cgi script in there. See the very bottom of the file. And make sure you look in ~/www/cgi-bin/.htaccess and not ~/www/.htaccess

**Comment by YLiLarry** [comment]
Works like a magic. Thanks Charlie.
This feature is finished.

**Pull request merged**